File: js/mco1/src/io-utils.ts
import process from "node:process";
import readline from "node:readline/promises";

export function printOrderedList<T extends { toString(): string }>(arr: readonly T[]): void {
  for (const [i, elm] of arr.entries()) {
    console.log(`[${i + 1}] ${elm.toString()}`);
  }
}

export async function prompt(msg: string): Promise<string> {
  const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

  const userIn = await rl.question(msg);

  rl.close();

  return userIn;
}

File: rust/mco1/src/io_utils.rs
use std::{
    fmt,
    io::{self, Write},
};

pub(crate) fn print_ordered_list<T: fmt::Display>(list: &[T]) {
    for (i, elm) in list.iter().enumerate() {
        println!("[{}] {elm}", i + 1);
    }
}

pub(crate) fn prompt(msg: &str) -> String {
    print!("{msg}");

    io::stdout().flush().expect("Failed to flush the output string...");

    let mut user_in = String::new();

    io::stdin()
        .read_line(&mut user_in)
        .expect("Failed to read the input string...");

    user_in.trim().to_string()
}
